Automattic/wp-calypso
# Make it possible to remove coupons from the cart

This PR is part of effort to improve everything coupons-related. We make it possible to remove a coupon from the cart after it was applied - which previously was impossible.

**Test plan:**

* Go to checkout page
* Click on "Have a coupon code?":

![](https://cdn-pro.dprcdn.net/files/acc_631815/CQMWXI)

* A coupon form should appear like this:

![](https://cdn-pro.dprcdn.net/files/acc_631815/FN2LAw)

* Enter an invalid coupon like "CODE20", coupon form should enter the "loading" state like this and then the error should appear:

<img width="273" alt="zrzut ekranu 2018-06-04 o 14 20 46" src="https://user-images.githubusercontent.com/205419/40917159-8522bbca-6802-11e8-855f-77936045088b.png">

* Find a valid coupon code in network admin and use it.
* A coupon should be applied and the form should look like this now:

<img width="292" alt="zrzut ekranu 2018-06-04 o 14 24 14" src="https://user-images.githubusercontent.com/205419/40917290-fb374bc8-6802-11e8-8884-63c88072c9bd.png">

* Click a "remove" link, coupon should be removed and the summary should look like this again:

<img width="289" alt="zrzut ekranu 2018-06-04 o 14 22 21" src="https://user-images.githubusercontent.com/205419/40917207-b4b93b7a-6802-11e8-9435-cf69b0b1b0e7.png">
